sagemaker-spark/pyspark_mnist/pyspark_mnist_kmeans.ipynb

```diff
@@ -0,0 +1,274 @@
+    "jars = sagemaker_pyspark.classpath_jars()\n",
+    "\n",
+    "classpath = \":\".join(sagemaker_pyspark.classpath_jars())\n",
+    "\n",
+    "# See the SageMaker Spark Github repo under sagemaker-pyspark-sdk\n",
+    "# to learn how to connect to a remote EMR cluster running Spark from a Notebook Instance.\n",
+    "spark = SparkSession.builder.config(\"spark.driver.extraClassPath\", classpath)\\\n",
+    "    .master(\"local[*]\").getOrCreate()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# replace this with your own region, such as us-east-1\n",
+    "region = 'us-east-1'\n",
+    "trainingData = spark.read.format('libsvm')\\\n",
+    "    .option('numFeatures', '784')\\\n",
+    "    .load('s3a://sagemaker-sample-data-{}/spark/mnist/train/'.format(region))\n",
+    "\n",
+    "testData = spark.read.format('libsvm')\\\n",
+    "    .option('numFeatures', '784')\\\n",
+    "    .load('s3a://sagemaker-sample-data-{}/spark/mnist/test/'.format(region))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading the Data\n",
+    "\n",
+    "In order to train and make inferences our input DataFrame must have a column of Doubles (named \"label\" by default) and a column of Vectors of Doubles (named \"features\" by default).\n",
+    "\n",
+    "Spark's LibSVM DataFrameReader loads a DataFrame already suitable for training and inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "trainingData.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training and Hosting a Model\n",
+    "Now we create a KMeansSageMakerEstimator, which uses the KMeans Amazon SageMaker Algorithm to train on our input data, and uses the KMeans Amazon SageMaker model image to host our model.\n",
+    "\n",
+    "Calling fit() on this estimator will train our model on Amazon SageMaker, and then create an Amazon SageMaker Endpoint to host our model.\n",
+    "\n",
+    "We can then use the SageMakerModel returned by this call to fit() to transform Dataframes using our hosted model.\n",
+    "\n",
+    "The following cell runs a training job and creates an endpoint to host the resulting model, so this cell can take up to twenty minutes to complete."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import random\n",
+    "from sagemaker_pyspark import IAMRole, S3DataPath\n",
+    "from sagemaker_pyspark.algorithms import KMeansSageMakerEstimator\n",
+    "\n",
+    "# replace this with your role ARN\n",
+    "kmeans_estimator = KMeansSageMakerEstimator(\n",
+    "    sagemakerRole=IAMRole(role),\n",
```

Comment:
does this mean `get_execution_role()` doesn't need to be called above?

Reply:
Removed this comment.